client/src/components/SettingsDialog.tsx

```diff
@@ -0,0 +1,86 @@
+import { Container, SegmentedControl, Select } from "@mantine/core";
+import { useClickOutside } from "@mantine/hooks";
+import React from "react";
+import useHTMLElementList from "../hooks/useHTMLElementList";
+import {
+  ThemeType,
+  usePreferencesActions,
+  useTheme,
+  useUnitPreferences,
+} from "../stores/PreferencesStore";
+import { stylesWithThemedBackgroundColor } from "../utils/styles";
+import {
+  MeasurementTypes,
+  DistanceUnits,
+  SpeedUnits,
+  AngleUnits,
+} from "../utils/units";
+
+interface SettingsDialogProps {
+  onClose?: () => void;
+}
+
+export default function SettingsDialog({ onClose }: SettingsDialogProps) {
+  const theme = useTheme();
+  const unitPreferences = useUnitPreferences();
+
+  const { updateUnitPreferences, updateTheme } = usePreferencesActions();
+
+  const { register, data: internalHTMLNodes } = useHTMLElementList();
+
+  useClickOutside(() => onClose?.(), undefined, internalHTMLNodes);
```

Comment:
There's an annoying bug with the implementation of the Mantine Select component where clicks on Select register as clicks external to the parent container. This manifested in the settings dialog closing whenever you'd select an option from any of the Select components we're using.

To get around this, I'm providing `internalHTMLNodes`, a list of elements that `useClickOutside` will treat as "internal" and anything else clicked is considered external. This list holds the container element as well as all of the Select refs, so choosing an option from a Select will NOT register as an external click and close the dialog.

Reply:
I originally had it implemented with a separate state variable for each node like mantine shows in their [docs](https://mantine.dev/hooks/use-click-outside/#multiple-nodes), but that wouldn't scale well if we had to add more selects here. That's why I added the `useHTMLElementList` hook. See the dif here: https://github.com/superballer365/golf-simulator/pull/7/commits/abe45f6b7d0e734241f5fa38aa0dda9fe12d5ee6

Reply:
Now that I think about it, I probably could've made our own custom `useClickOutside` built on top of Mantine's that returns the `register` function and it will let you easily add multiple nodes to the "internal" nodes... I think I'll do that